homie-influx/src/main.rs

```diff
@@ -0,0 +1,388 @@
+                        device_id,
+                        node_id,
+                        property_id,
+                        value,
+                        fresh,
+                    } => {
+                        log::trace!(
+                            "{}/{}/{} = {} ({})",
+                            device_id,
+                            node_id,
+                            property_id,
+                            value,
+                            fresh
+                        );
+                        if fresh {
+                            send_property_value(
+                                controller.as_ref(),
+                                &influx_db_client,
+                                device_id,
+                                node_id,
+                                property_id,
+                            )
+                            .await;
+                        }
+                    }
+                    _ => {
+                        log::info!("Event: {:?}", event);
+                    }
+                }
+            }
+        }
+    })
+}
+
+async fn send_property_value(
+    controller: &HomieController,
+    influx_db_client: &Client,
+    device_id: String,
+    node_id: String,
+    property_id: String,
+) {
+    if let Some(device) = controller.devices().get(&device_id) {
+        if let Some(node) = device.nodes.get(&node_id) {
+            if let Some(property) = node.properties.get(&property_id) {
+                if let Some(point) =
+                    point_for_property_value(device, node, property, SystemTime::now())
+                {
+                    // TODO: What should rp be?
+                    // TODO: Handle errors
+                    influx_db_client
+                        .write_point(point, INFLUXDB_PRECISION, None)
+                        .await
+                        .unwrap();
+                }
+            }
+        }
+    }
+}
+
+/// Convert the value of the given Homie property to an InfluxDB value of the appropriate type, if
+/// possible. Returns None if the datatype of the property is unknown, or there was an error parsing
+/// the value.
+fn influx_value_for_homie_property(property: &Property) -> Option<Value> {
+    let datatype = property.datatype?;
+    Some(match datatype {
+        Datatype::Integer => Value::Integer(property.value().ok()?),
+        Datatype::Float => Value::Float(property.value().ok()?),
+        Datatype::Boolean => Value::Boolean(property.value().ok()?),
+        _ => Value::String(property.value.to_owned()?),
+    })
+}
+
+/// Construct an InfluxDB `Point` corresponding to the given Homie property value update.
+fn point_for_property_value(
+    device: &Device,
+    node: &Node,
+    property: &Property,
+    timestamp: SystemTime,
+) -> Option<Point> {
+    let datatype = property.datatype?;
+    let value = influx_value_for_homie_property(property)?;
+
+    let mut point = Point::new(&datatype.to_string())
```

Comment:
So this will be one of Integer, Float, Boolean, String, Enum, Color? Seems suspicious but it might just work ;-).

Reply:
Lowercase, but yes. See my comment on the PR for an example of how the data looks like in InfluxDB.